eslint-rules/ts-no-use-before-define.js

```diff
@@ -0,0 +1,288 @@
+// Added here until https://github.com/nzakas/eslint-plugin-typescript/pull/34
```

Comment:
Part of the reason to have lint rules like this is to catch errors more quickly — the linter is much faster than compiling all of the code using webpack/TS.